crates/stages/src/stages/execution.rs

```diff
@@ -143,14 +153,15 @@ impl ExecutionStage {
 
             trace!(target: "sync::stages::execution", number = block_number, txs = transactions.len(), "Executing block");
 
-            let changeset = reth_executor::executor::execute_and_verify_receipt(
-                &Block { header, body: transactions, ommers, withdrawals },
-                td,
-                Some(signers),
-                &self.chain_spec,
-                &mut state_provider,
-            )
-            .map_err(|error| StageError::ExecutionError { block: block_number, error })?;
+            // Configure the executor to use the current state.
+            let mut executor = self.executor.with_db(&mut state_provider);
```

Comment:
this looks like a redundant call to make on each iteration. shouldn't this be above along with state provider initialization?

Reply:
I think the primary issue rn is that `SubState` is just a type alias for some wrapper around a `StateProvider` and revm's `CacheDB`, so there is no expiry/rotation of the cache. If you executed from block 0 to 10 million you would end up having the entire state in memory, so for now, we have to do this I guess

Reply:
cc @rakita if you have any thoughts on this